### ISSUE TITLE
Error: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.

### ISSUE THREAD

**Issue by carrbrpoa** [Esri/esri-leaflet#667]
Despite of [issue 455](https://github.com/Esri/esri-leaflet/issues/455), I'm still having similar problem.

I'm toggling between my available baselayers (with or without overlays) and sometimes this error comes:

![error_toggling_baselayers](https://cloud.githubusercontent.com/assets/10212618/10693881/dad72c38-7979-11e5-8341-af6b2156918c.png)

And the line pointed in ```BasemapLayer.js``` is:

```javascript
onRemove: function(map){
      // check to make sure the logo hasn't already been removed
      if(this._logo && this._logo._container){
        map.removeControl(this._logo); // Line 232
        map._hasEsriLogo = false;
      }
      (...)
```

Here's a plnkr: http://plnkr.co/edit/jyXgske9YXYRViqLamBc?p=preview

It's a simplified version of my app, without overlays. See that you just need to start switching the baselayers and in a while the error occurs.

Sometimes it just doesn't loads the selected baselayer due to the error.

Any ideas?

Thanks in advance

**Comment by jgravois** [comment]
thanks for reporting.  i can't reproduce the error when targeting `master` and leaflet `1.0.0-beta.2`, so i'm tempted to just ask you to upgrade to `2.0.0-beta.6` if the chatter in the console bothers you.

```js
var osm = L.tileLayer('http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png');
var gray = L.esri.basemapLayer('Gray');
var dark = L.esri.basemapLayer('DarkGray');
var topo = L.esri.basemapLayer('Topographic');

var baseMaps = {
  "osm": osm,
  "gray": gray,
  "dark gray": dark,
  "topographic": topo
};

var map = L.map('map', {
    center: [39.73, -104.99],
    zoom: 3
});

L.control.layers(baseMaps).addTo(map);
```

**Comment by carrbrpoa** [comment]
Hey @jgravois, that behaviour was actually breaking the code, not just bothering ;)
But ok, changing to last beta versions of everything (leaflet and esri-leaflet) worked.

Thanks

**Comment by jgravois** [comment]
ok.  weird that i didn't see the same thing, but either way, good to know that you were able to upgrade conveniently.

**Comment by patrickarlt** [comment]
@jgravois reopening this to ask @carrbrpoa a question.

@carrbrpoa What versions of Leaflet/Esri Leaflet were you using that produced the error? I just want to make sure we don't have an error on `master` or the 1.0.0 version right now.

**Comment by carrbrpoa** [comment]
Hey @patrickarlt. Sorry, I overwritted the plnkr. This is a *new plnkr* with the versions that produced the error (leaflet 0.7.7 or 0.7.5 and esri-leaflet 1.0.0):

http://plnkr.co/edit/saRmlbilSA6mAIDpgIXI?p=preview

**Comment by jgravois** [comment]
thx @carrbrpoa, I'll figure out what's going on and get a patch in.

**Comment by patrickarlt** [comment]
Thanks @jgravois assigning to you.

**Comment by jgravois** [comment]
i've got this problem fixed in a branch called [`container-error`](https://github.com/Esri/esri-leaflet/tree/container-error).  it makes sense to me that i can tag `1.0.1` as a patch directly from my commit without merging to master, and i assume that npm will fold the release in appropriately too but i'll wait a bit just in case anyone has any words of caution.

**Comment by patrickarlt** [comment]
@jgravois yup. Release process should be.

1. Bump `version` in package.json
2. Update the changelog
3. Commit and push to `container-error`

I think right now only I have permissions to do an NPM release unless you also have access to the esri_dev account so I can take it from there once you do those steps.

**Comment by jgravois** [comment]
> unless you also have access to the `esri_dev` account

i've got the keys.

while i was riding my bike to work this morning i thought of one more thing.  what do we do if/when we need to release 1.0.2?  if i delete my branch after tagging it will no longer be in the repository tree.  should we maintain a long running 1.x branch going forward?

**Comment by jgravois** [comment]
> 2. Update the changelog

just a note that since we aren't actually ever going to merge `1.x` bug fixes back into master, the update to the changelog has to happen in a commit in an entirely seperate Pull Request.

**Comment by jgravois** [comment]
i've verified the problem is fixed in [1.0.1](https://github.com/Esri/esri-leaflet/releases/tag/v1.0.1).  @carrbrpoa thank you **so** much for taking the time to point out the problem and provide a repro case.

just as an FYI, it was necessary to run the command below to ensure that the patch i tagged isn't considered the 'latest' release on npm
```
npm dist-tag add esri-leaflet@2.0.0-beta.6 latest
```
in the future, i'll consider setting a different [`--tag`](https://docs.npmjs.com/cli/publish) manually when publishing or adding [`publishConfig`](https://github.com/npm/npm/issues/6778#issuecomment-75447332) to the package.json.

**Comment by carrbrpoa** [comment]
Thanks for the efforts guys :100:
